src/GitVersionCore/GitRepoInformation/GitRepoMetadata.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+
+namespace GitVersion.GitRepoInformation
+{
+    public class GitRepoMetadata
+    {
+        public GitRepoMetadata(
+            List<MTag> tags, MBranch currentBranch, MBranch master,
+            List<MBranch> releaseBranches)
+        {
+            Tags = tags;
+            MasterBranch = master;
+            CurrentBranch = currentBranch;
+            ReleaseBranches = releaseBranches;
+        }
+
+        // Wonder if this can be 'mainline'
+        public MBranch MasterBranch { get; }
+        public List<MBranch> ReleaseBranches { get; }
+        public MBranch CurrentBranch { get; }
+        public List<MTag> Tags { get; private set; }
+    }
+
+    public class MTag
```

Comment:
Rather than aliasing in a bunch of files which reference both this namespace and libgit2 the metadata related ones are prefixed with M. I expect this to go at some point